server/daemon/run_insecure_dev_server.sh

```diff
@@ -29,7 +29,9 @@ if [ ! -f "${CONFIG_FILE}" ]; then
     exit 1
 fi
 
-pushd "${SCRIPT_DIR}" > /dev/null 2>&1
+# Save current directory and change to script directory without pushd
+OLD_DIR=$(pwd)
+cd "${SCRIPT_DIR}" > /dev/null 2>&1
```

Comment:
I don't think the `> /dev/null 2>&1` is required here. The only time cd should print anything is if the directory doesn't exist in which case swallowing the error is probably not great for debugging.

Reply:
That probably should be a `cd "${SCRIPT_DIR}" || exit 1`